src/citrine/_serialization/properties.py

```diff
@@ -52,26 +80,32 @@ def serialized_types(self) -> typing.Union[SerializedType, typing.Tuple[Serializ
     def _error_source(self, base_class: type) -> str:
         """Construct a string of the base class name and the parameter that failed."""
         if base_class is not None:
-            return 'for {}:{}'.format(base_class.__name__, self.serialization_path)
+            return ' for {}:{}'.format(base_class.__name__, self.serialization_path)
         elif self.serialization_path:
-            return 'for {}'.format(self.serialization_path)
+            return ' for {}'.format(self.serialization_path)
         else:
             return ''
 
     def serialize(self, value: DeserializedType,
-                  base_class: Optional[type] = None) -> SerializedType:
+                  base_class: typing.Optional[type] = None) -> SerializedType:
         if not isinstance(value, self.underlying_types):
             base_name = self._error_source(base_class)
-            raise ValueError('{} is not one of valid types: {} {}'.format(
-                value, self.underlying_types, base_name))
+            raise ValueError(
+                f'{type(value)} {value} is not one of valid types: '
+                f'{self.underlying_types}{base_name}'
+            )
         return self._serialize(value)
 
     def deserialize(self, value: SerializedType,
-                    base_class: Optional[type] = None) -> DeserializedType:
+                    base_class: typing.Optional[type] = None) -> DeserializedType:
         if not isinstance(value, self.serialized_types):
+            if isinstance(value, self.underlying_types):
+                return value  # Don't worry if it was already deserialized
```

Comment:
In what situation are we trying to deserialize an already-deserialized object?

Reply:
The `flatten` utility from gemd-python only partly serializes its objects when dissecting them, the validate_templates method in citrine-python does something similar, and many tests use build combined with Objects in the incoming values stream.

Maybe not ideal, but correcting this would make this PR substantially larger.  Not a new behavior.